python/foglamp/services/core/scheduler/scheduler.py

```diff
@@ -1078,6 +1077,50 @@ def _schedule_row_to_schedule(cls,
             self._schedule_first_task(schedule_row, now)
             self._resume_check_schedules()
 
+    async def ping_service(self, service):
+        _MAX_ATTEMPTS = 5
+        """Number of max attempts for finding a heartbeat of service"""
+        attempt_count = 1
+        """Number of current attempt to ping url"""
+
+        url_ping = "{}://{}:{}/foglamp/service/ping".format(service._protocol,
+                                                            service._address,
+                                                            service._management_port)
+        async with aiohttp.ClientSession() as session:
+            while attempt_count < _MAX_ATTEMPTS + 1:
+                try:
+                    async with session.get(url_ping) as resp:
+                        res = await resp.json()
+                        if res["uptime"] is not None:
+                            break
+                except:
+                    attempt_count += 1
+                    time.sleep(1)
+            if attempt_count <= _MAX_ATTEMPTS:
+                self._logger.info('Ping received for Service %s id %s at url %s', service._name, service._id, url_ping)
+                return True
+        self._logger.error('Ping not received for Service %s id %s at url %s attempt_count %s', service._name, service._id,
+                      url_ping, attempt_count)
+        return False
+
+    async def shutdown_service(self, service):
```

Comment:
again, utils? I believe each scheduled (microservice) plugin shall need it. 
And/Or core server shutdown (just to ensure all microservices are shutdown except storage and core) as in Pr for FOGL-653